Makefile

```diff
@@ -35,31 +35,24 @@ help: ## Show this help.
 ---------------- : ## ----------------
 
 .PHONY: setup
-setup: ## Set up a development environment
-ifeq ($(OS),Linux)
-	changes="`sudo snap install --no-wait codespell`"
-	changes="$$changes `sudo snap install --no-wait ruff`"
-	changes="$$changes `sudo snap install --no-wait shellcheck`"
-	changes="$$changes `sudo snap install --classic --no-wait astral-uv`"
-	for change in $$changes; do
-		snap watch $$change
-	done
-else ifeq ($(OS),Windows_NT)
-	pipx install uv
-	choco install shellcheck
-else ifeq ($(OS),Darwin)
-	brew install uv
-	brew install shellcheck
-endif
-ifneq ($(OS),Linux)
-	uv tool install codespell
-	uv tool install ruff
-	uv tool update-shell
-endif
+setup: install-uv setup-precommit ## Set up a development environment
+	uv sync --frozen --all-extras
+
+.PHONY: setup-tests
+setup-tests: install-uv  ##- Set up a testing environment without linters
+	uv sync --frozen
+
+.PHONY: setup-lint
+setup-lint: install-uv install-shellcheck  ##- Set up a linting-only environment
+	uv sync --frozen --no-dev --no-install-workspace --extra lint --extra types
+
+.PHONY: setup-docs
+setup-docs: install-uv  ##- Set up a documentation-only environment
+	uv sync --frozen --no-dev --no-install-workspace --extra docs
 
-.PHONY: setup-pre-commit
-setup-precommit:  ## Set up pre-commit hooks in this repository.
-ifeq (, $(shell which pre-commit))
+.PHONY: setup-precommit
+setup-precommit: install-uv  ##- Set up pre-commit hooks in this repository.
+ifeq ($(shell which pre-commit),)
 	uv tool install pre-commit
 endif
 	pre-commit install
```

Comment:
I'd remove the dashes here - change the comments to `## Set up whatever`.  This way they don't appear as subordinate targets in the help output.

Reply:
I'm conflicted on `setup-precommit`. On the other ones I want to leave them as is because while not technically subordinate targets, they do a subset of what `setup` does. For `setup-precommit` I wonder if I should just make `setup` depend on it.